test/test_output_writers.jl

```diff
@@ -88,27 +90,61 @@ function test_dependency_adding(model)
     return nothing
 end
 
+function test_creating_and_appending(model, output_writer)
+
+    simulation = Simulation(model, Δt=1, stop_iteration=5)
+    output = fields(model)
+    filename = "test_caa.h"
+
+    # Create a simulation with `overwrite_existing = true` and run it
+    simulation.output_writers[:writer] = output_writer(model, output,
+                                                       filename = filename,
+                                                       schedule = IterationInterval(1),
+                                                       overwrite_existing = true)
```

Comment:
Having both writers be extension-agnostic lets us write general stuff like this :)

Reply:
That's how most software works for example google docs:

![image](https://user-images.githubusercontent.com/15271942/162793709-cb4675f1-d0c5-4cac-8980-2436b7bc2564.png)

gives `DocumentName.docx`, `DocumentName.odt`, etc.

Reply:
> I guess if we append the extension automatically then we'd write
> 
> ```julia
>     filename = "test_caa"
> ```
> 
> and attain equal generality?

I think if we're going to enforce the extension, it's best that we ask for the full filename and check if the extension is appropriate, rather than asking for the prefix and appending the extension. I think it's clearer for the user and makes for shorter code too.